### PR TITLE
[Impeller] Implement a Vulkan backend.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This is WIP and the backend is NOT functional yet. It wires up the compiler and
basic context creation and pipeline state object generation. In the process of
wiring this up, I ran into a few action items for us to tackle in the higher
level API so this backend can be fully setup. All issues are minor and
referenced inline.

* https://github.com/flutter/flutter/issues/106376
* https://github.com/flutter/flutter/issues/106377
* https://github.com/flutter/flutter/issues/106378
* https://github.com/flutter/flutter/issues/106379
* https://github.com/flutter/flutter/issues/106380

Beyond the issues identified, there are no other blockers or unknowns. I expect
this could be done with about a weeks worth of effort.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
I am ambivalent about actually landing this incomplete backend. IMO the important bits are the linked issues that describe minor changes to the higher level API. We should incorporate them soon so that a Vulkan or D3D12 backend keeps being a natural fit for Impeller should we choose to have this backend and we don't paint ourselves into a corner.

File: impeller/blobcat/blob_writer.cc [hunk 1]
```diff
@@ -31,7 +31,7 @@ bool BlobWriter::AddBlobAtPath(const std::string& std_path) {
     return false;
   }
 
-  if (path.extension() != ".gles") {
+  if (path.extension() != ".gles" && path.extension() != ".vkspv") {
```
Comment (dnfield) [inline impeller/blobcat/blob_writer.cc:34]: I may be missing context here but does this have any implication for the fragment shader API?

Comment (chinmaygarde) [reply impeller/blobcat/blob_writer.cc:34]: No. This is just an intermediate in the gen directory. The Impeller FragmentProgram API will take these blobs, slap some metadata on it and embed it in the final binary.

**Comment by bdero** [comment]
Nice!

**Comment by chinmaygarde** [comment]
The takeaways from this prototype were the issues already filed. If we need this backend, we can revive it with little effort. Closing and putting this in the icebox.

**Pull request closed without merging**